MoltenVK/MoltenVK/Commands/MVKCommandEncoderState.h

```diff
@@ -387,11 +388,12 @@ class MVKResourcesCommandEncoderState : public MVKCommandEncoderState {
         T db = b;   // Copy that can be marked dirty
         MVKCommandEncoderState::markDirty();
         bindingsDirtyFlag = true;
-        db.isDirty = true;
+        db.markDirty();
 
         for (auto iter = bindings.begin(), end = bindings.end(); iter != end; ++iter) {
             if( iter->index == db.index ) {
-                *iter = db;
+                if (!areEqual(&db, iter))
+                    *iter = db;
                 return;
             }
         }
```

Comment:
Per the comment below by @cdavis5e, regarding side effects in `areEqual()`, I'm wondering if it might be a  good idea to merge this logic into something like an `updateIfReplaced(b)` function on the binding structs, and call `areEqual()` and `markDirty()`, plus handle the additional specialized buffer logic for `justOffset` within that `updateIfReplaced(b)` function. Then we'd end up with something here like:

```
for (auto iter = bindings.begin(), end = bindings.end(); iter != end; ++iter) {
    if (iter->updateIfReplaced(b))
        return;
}
```

with no need for the `db` copy, because the `updateIfReplaced()` function in `*iter` would call it's own `markDirty()`. And I've assumed `areEqual() could presumably include the `index == b.index` check too.